cmake/sources.cmake

```diff
@@ -0,0 +1,98 @@
+# Globbing is a source of bugs, sources are named explicitly here.
+# TODO look into file sets.
+set( GODOTCPP_SOURCES
```

Comment:
I understand that globbing can introduce issues, however, scons is using globbing, and this will mean folks who add new source code files will need to know to do something for cmake that they don't need to do for scons, which could lead to this getting missed and breaking the cmake build. It might be better to use globbing in cmake too to avoid that?

Reply:
I can change it, however file globbing breaks the depedency chain forcing the target to be rebuilt every time because things might have changed.

The godot-cpp static sources are something we manage, so having a list of them isn't a big deal to maintain.

The generated sources however are a different problem. I could, make the sources list generated from the file list that the code-gen uses. does that satisfy?

Reply:
@dsnopek I just had another look at the source inclusion, and generated sources are already taken from the generation phase and included in the build without the need for the globbing. I see there is a question mark at the end of your statement.. it might be better?  There are long [discussions ](https://stackoverflow.com/questions/1027247/is-it-better-to-specify-source-files-with-glob-or-each-file-individually-in-cmak) on the downsides of globbing when using cmake, especially for large distributed projects with many people working on it.

Reply:
I've replaced the source list with globbing as requested.